dev/io.openliberty.http.monitor/src/io/openliberty/http/monitor/HttpStatsMonitor.java

```diff
@@ -229,8 +272,37 @@ private synchronized HttpStats initializeHttpStat(String key, HttpStatAttributes
 
 		HttpStats httpMetricStats = new HttpStats(statAttri);
 		HttpConnByRoute.put(key, httpMetricStats);
+		
+		/*
+		 * null means from server.
+		 * Specifically splash page.
+		 * 
+		 * Add to appName -> stat cache
+		 */
+		if (appName != null) {
+			appNameToStat.compute(appName, (appNameKey, currValSet) -> {
+				if (currValSet == null) {
+					HashSet<String> hs = new HashSet<String>();
+					hs.add(key);
+					return hs;
+				} else {
+					currValSet.add(key);
+					return currValSet;
+				}
+			});
+		}
+		
 		return httpMetricStats;
 	}
+	
+	
+	public void removeStat(String appName) {
+		Set<String> retSet = appNameToStat.get(appName);
+		if (retSet != null) {
+			retSet.stream().forEach( statName -> HttpConnByRoute.remove(statName));
```

Comment:
```suggestion
			for (String statName : retSet) {
				HttpConnByRoute.remove(statName));
			}
```
There is no value in using stream in this case and it will just perform so much worse.